src/containers/AdminCampaignEdit/components/StartCampaignButton.tsx

```diff
@@ -38,11 +47,36 @@ export const StartCampaignButton: React.FC<StartCampaignButtonProps> = (
 
   const tooltipText = requiresApproval ? "Superadmin approval required" : "";
 
+  const campaignInteractionSteps =
+    campaignStepsData?.campaign?.interactionSteps ?? [];
+
+  const campaignHasSurveyResponses = campaignInteractionSteps.length > 1;
+
+  const handleConfirmWarningAndStartCampaign = useCallback(() => {
+    setWarningConfirmed(true);
+    setConfirmDialogOpen(false);
+    startCampaign({ variables: { campaignId } });
+  }, [startCampaign, campaignId, setWarningConfirmed, setConfirmDialogOpen]);
+
+  const handleCloseDialog = useCallback(() => {
+    setConfirmDialogOpen(false);
+  }, [setConfirmDialogOpen]);
```

Comment:
Nit: I think this doesn't need a dependency or maybe even fine to render this without `useCallback()` ?
```suggestion
  const handleCloseDialog = useCallback(() => {
    setConfirmDialogOpen(false);
  }, []);
```